screens/Budget/CreateBudgetScreen.tsx

```diff
@@ -64,24 +64,24 @@ export default function CreateExpenseScreen({ navigation }: RootStackScreenProps
         }
     }
 
-    function handleSubmit() {
-        // submitNewBudget();
-        navigation.goBack();
+    function handleBudgetCreation() {
+        if (amount == 0) {
+            alert("Budgeted amount can't be zero")
+        } else if (budget == null || budget.id == null) {
+            alert("Parent budget not found")
+        } else if (categoryId == null) {
+            alert("Please select a catgeory")
+        } else {
+            createBudget()
+        }
+
     }
-    console.log("calivn", amount)
+
     return (
         <View style={styles.screen}>
             <View style={styles.amountInputContainer}>
                 <View style={styles.dollarSignAndAmountInput}>
                     <Text style={styles.dollarSign}>$</Text>
-                    {/* <AdaptiveTextInput
-                        keyboardType="numeric"
-                        style={{ fontSize: 50 }}
-                        charWidth={30}
-                        value={amount}
-                        onChangeText={handleAmountChange}
-                        onBlur={handleAmountBlur}>
-                    </AdaptiveTextInput> */}
                     <InputDecimal callback={setAmount} type="text" placeholder='amount' />
```

Comment:
Change this to our custom amount input component. I will fix it to make it more intuitive to use.